### PR TITLE
Update pf component example 

### PR DESCRIPTION
**Pull request by alainli0928**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this
page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When using the `load_component` function and the flow YAML specification, your flow is automatically transformed into a __[parallel component](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-use-parallel-job-in-pipeline?view=azureml-api-2&tabs=cliv2#why-are-parallel-jobs-needed)__. This parallel component is designed for large-scale, offline, parallelized processing with efficiency and resilience. Here are some key features of this auto-converted component:\n",
+    "\n",
+    " - Pre-defined input and output ports:\n",
+    "\n",
+    "    ![prompt flow base component image](../../../docs/media/cloud/flow-in-pipeline/pf-base-component.png)\n",
+    "    | port name  |  type  | description |\n",
```
Comment (elliotzh) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:133]: image
Note sure about the reason but seems that there are some issues

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+    "\n",
+    "Before you begin, consider the following prerequisites:\n",
+    "- Introduction to Azure ML Platform:\n",
+    "    - [Core site of Azure ML platform](https://learn.microsoft.com/en-us/azure/machine-learning/overview-what-is-azure-machine-learning?view=azureml-api-2).\n",
+    "    - Understand what [Azure ML pipelines](https://learn.microsoft.com/en-us/azure/machine-learning/concept-ml-pipelines?view=azureml-api-2) and [component](https://learn.microsoft.com/en-us/azure/machine-learning/concept-component?view=azureml-api-2) are.\n",
+    "- Azure cloud setup:\n",
+    "    - An Azure account with an active subscription - [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F)\n",
+    "    - Create an Azure ML resource from Azure portal - [Create a Azure ML workspace](https://ms.portal.azure.com/#view/Microsoft_Azure_Marketplace/MarketplaceOffersBlade/searchQuery/machine%20learning)\n",
+    "    - Connect to your workspace then setup a basic computer cluster - [Configure workspace](../../configuration.ipynb)\n",
+    "- Local environment setup:\n",
```
Comment (elliotzh) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:28]: Would you like to mention some pre-requirements here? Like the azure-ai-ml version, `flow.tools.json` must be generated (via `pf flow validate`), and so on?

Comment (elliotzh) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:28]: image

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+    "If you’ve already authored a flow using the Promptflow SDK or portal, you can locate the flow.dag.yaml file within the flow folder. This YAML specification is essential for loading your flow into an Azure ML component."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "flow_component = load_component(\"../../flows/standard/web-classification/flow.dag.yaml\")"
```
Comment (elliotzh) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:121]: would you like to mention how azureml environment is supported for Flow component? You can refer to [this doc](https://microsoft.github.io/promptflow/cloud/azureai/use-flow-in-azure-ml-pipeline.html?highlight=flow+pipeline#environment-of-the-component).

**Comment by elliotzh** [comment]
we have a document at promptflow side, too: [link](https://microsoft.github.io/promptflow/cloud/azureai/use-flow-in-azure-ml-pipeline.html?highlight=flow+pipeline#environment-of-the-component).

Feel free to update that one if you think necessary

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb (hunk 1 above)
Comment (alainli0928) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:133]: Good catch. It's very strange as it works well with VSCode... Let me try to fix it.

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb (hunk 3 above)
Comment (alainli0928) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:121]: Good point. Will add more content here.

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 4]
```diff
@@ -0,0 +1,613 @@
+{
+ "cells": [
```
Comment (wangchao1230) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:2]: The folder name seems a bit odd: run-scalable-flow-with-pipeline/pipeline.ipynb
We dont have 'scalabale flow', shall we just name this run-flow-with-pipeline

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 5]
```diff
@@ -0,0 +1,613 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Why use Azure machine learning(ML) pipelines to run your flows on the cloud?\n",
+    "In real-world scenarios, flows serve various purposes. For example, consider a flow designed to evaluate the relevance score for a communication session between humans and agents. Suppose you want to trigger this flow every night to assess today’s performance and avoid peak hours for LLM (Language Model) endpoints. In this common scenario, people often encounter the following needs:\n",
```
Comment (wangchao1230) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:8]: make our reference doc point to this example?

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+    "- Scalability and Efficiency: Requiring a scalable, efficient, and resilient platform to ensure success.\n",
+    "- Automations: Automatically triggering batch flows when upstream data is ready or at fixed intervals.\n",
+    "\n",
+    "__Azure ML pipelines__ address all these offline requirements effectively. With the integration of prompt flows and Azure ML pipeline, flow users could very easily achieve above goals and in this tutorial, you can learn:\n",
+    "- How to use python SDK to automatically convert your flow into a 'step' in Azure ML pipeline.\n",
+    "- How to feed your data into pipeline to trigger the batch flow runs.\n",
+    "- How to build other pipeline steps ahead or behind your prompt flow step. e.g. data preprocessing or result aggregation.\n",
+    "- How to setup a simple scheduler on my pipeline.\n",
+    "- How to deploy pipeline to an Azure ML batch endpoint. Then I can invoke it with new data when needed.\n",
+    "\n",
```
Comment (wangchao1230) [inline examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:19]: shall we also revamp the examples here？ https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/pipelines/1l_flow_in_pipeline/flow_in_pipeline.ipynb

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb (hunk 5 above)
Comment (alainli0928) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:8]: Do you mean the doc at here: https://microsoft.github.io/promptflow/index.html
If so, I'll plan the update later

Comment (alainli0928) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb:8]: And a question for doc & repo, which one would be the first entrance from customer?

File: examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb (hunk 4 above)
Comment (alainli0928) [reply examples/tutorials/run-scalable-flow-with-pipeline/pipeline.ipynb]: Fixed

File: examples/tutorials/run-flow-with-pipeline/pipeline.ipynb [hunk 7]
```diff
@@ -0,0 +1,628 @@
...
+    "    - Understand what [Azure ML pipelines](https://learn.microsoft.com/en-us/azure/machine-learning/concept-ml-pipelines?view=azureml-api-2) and [component](https://learn.microsoft.com/en-us/azure/machine-learning/concept-component?view=azureml-api-2) are.\n",
+    "- Azure cloud setup:\n",
+    "    - An Azure account with an active subscription - [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F)\n",
+    "    - Create an Azure ML resource from Azure portal - [Create a Azure ML workspace](https://ms.portal.azure.com/#view/Microsoft_Azure_Marketplace/MarketplaceOffersBlade/searchQuery/machine%20learning)\n",
+    "    - Connect to your workspace then setup a basic computer cluster - [Configure workspace](../../configuration.ipynb)\n",
+    "- Local environment setup:\n",
+    "    - A python environment\n",
+    "    - Installed Azure Machine Learning Python SDK v2 - [install instructions](../../../README.md) - check the getting started section and make sure version of 'azure-ai-ml' is higher than `1.12.0`\n",
+    "    - "
```
Comment (elliotzh) [inline examples/tutorials/run-flow-with-pipeline/pipeline.ipynb:31]: NIT, extra line

File: examples/tutorials/run-flow-with-pipeline/pipeline.ipynb [hunk 8]
```diff
@@ -0,0 +1,628 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Submit the pipeline job to your workspace\n",
+    "pipeline_job_run = ml_client.jobs.create_or_update(\n",
+    "    pipeline_job_def, experiment_name=\"Single_flow_component_pipeline_job\"\n",
+    ")\n",
+    "pipeline_job_run"
```
Comment (elliotzh) [inline examples/tutorials/run-flow-with-pipeline/pipeline.ipynb:297]: do a stream here to make sure the job may complete?

**Review by elliotzh — DISMISSED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by elliotzh — APPROVED** [review]

**Pull request merged**